src/app/features/public-form/public-form.component.ts

```diff
@@ -110,9 +110,35 @@ export class PublicFormComponent<E extends Entity> implements OnInit {
     }
     this.formConfig = migratePublicFormConfig(this.formConfig);
     this.fieldGroups = this.formConfig.columns;
+    this.handlePrefilledFields();
+
     await this.initForm();
   }
 
+  private handlePrefilledFields() {
+    if (!this.formConfig.prefilledFields?.length) {
+      return;
+    }
+
+    this.formConfig.prefilledFields.forEach((item) => {
+      if (item.id) {
+        const newField: FormFieldConfig = {
+          id: item.id,
+          defaultValue: item.defaultValue || null,
+          hideFromForm: item.hideFromForm ?? true,
+        };
+        const fieldGroup = this.fieldGroups.find((group) =>
+          group.fields.some((field) => field === item.id),
+        );
+        if (fieldGroup) {
+          fieldGroup.fields.push(newField);
+        } else {
+          this.fieldGroups.push({ fields: [newField] });
+        }
```

Comment:
I don't think it makes a difference where we add the field.

Currently, even when finding the field is already there, you still add it a second time, just to the same group, don't you? This seems to work, although I think overwriting only the defaultValue for an existing field would be cleaner.

Please also add two unit tests:
- a) prefilled has a field that is not part of visible fields yet (should get added hidden)
- b) prefilled has a field that is also in visible fields (should take on the new defaultValue)

Reply:
I have adjusted the tests a bit for clarity (the new PublicFormConfig was not being used anyway, for example  ...)